### PR TITLE
Add knocked rooms to sync response struct

### PR DESCRIPTION
**Pull request by Fizzadar**
Adds knocked rooms to sync response spec.

### CODE REVIEW

File: responses.go [hunk 1]
```diff
@@ -253,7 +253,7 @@ type RespSync struct {
 
 	DeviceLists    DeviceLists       `json:"device_lists"`
 	DeviceOTKCount OTKCount          `json:"device_one_time_keys_count"`
-	FallbackKeys   []id.KeyAlgorithm `json:"device_unused_fallback_key_types"`
+	FallbackKeys   []id.KeyAlgorithm `json:"device_unused_fallback_key_types,omitempty"`
```
Comment (tulir) [inline responses.go:256]: That looks a bit sus

Comment (Fizzadar) [reply responses.go:256]: It’s optional in the spec I believe, for non e2ee clients I’d guess?

Comment (tulir) [reply responses.go:256]: It's required in the spec, although it won't exist in servers that don't support the spec version that added it. Is the omitempty needed for something?

File: responses.go [hunk 2]
```diff
@@ -338,19 +339,26 @@ func (sir SyncInvitedRoom) MarshalJSON() ([]byte, error) {
...
-	Servers []string  `json:"servers"`
-}
+type (
+	RespAliasCreate  struct{}
+	RespAliasDelete  struct{}
+	RespAliasResolve struct {
+		RoomID  id.RoomID `json:"room_id"`
+		Servers []string  `json:"servers"`
+	}
+)
```
Comment (sumnerevans) [inline responses.go:360]: does this formatting change actually do anything besides group these together?

If we are grouping `RespAlias*` together, why is `RespAliasList` not included here?

File: responses.go (hunk 1 above)
Comment (Fizzadar) [reply responses.go:256]: Oh my bad - doesn't show required under the sync spec but does under OTK, will remove this as was only making the change for spec compliance.

File: responses.go (hunk 2 above)
Comment (Fizzadar) [reply responses.go:360]: Woa! Not sure how this slipped in think this was `gofumpt` sneaking this in here let me fix that.

**Comment by Fizzadar** [comment]
OK I have fixed this mess, just the knocked room addition now.

**Review by tulir — APPROVED** [review]

**Review by sumnerevans — APPROVED** [review]

**Pull request merged**